### PR TITLE
SignUp screen

### PR DESCRIPTION
**Pull request by miguelhenaoa**
## Summary

- Screen and sign up form
- Update React Hook Form library

## Screenshots

image

## Trello Card

https://trello.com/c/OYU7tQFY

### CODE REVIEW

File: src/screens/SignUp/index.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            {...register('lastName', { required: 'Campo requerido' })}
+            label="Apellido"
+            type="text"
+            error={errors.lastName?.message}
+          />
+          <Input
+            {...register('email', {
+              required: 'Campo requerido',
+              pattern: {
+                value: /\S+@\S+\.\S+/,
```
Comment (TaloMontes) [inline src/screens/SignUp/index.tsx:39]: Add const with this regex

File: src/screens/SignUp/index.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            })}
+            label="Email"
+            type="email"
+            error={errors.email?.message}
+          />
+          <Input
+            {...register('password', {
+              required: 'Campo requerido',
+              pattern: {
+                value: /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
```
Comment (TaloMontes) [inline src/screens/SignUp/index.tsx:51]: same

File: src/scss/buttons.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@import './variables/colors';
+
+%btn {
+  height: 32px;
+  width: 252px;
+  border-radius: 5px;
```
Comment (TaloMontes) [inline src/scss/buttons.scss:6]: sort alphabetically

File: src/screens/SignUp/index.tsx (hunk 1 above)
Comment (miguelhenaoa) [reply src/screens/SignUp/index.tsx]: Done!

File: src/scss/buttons.scss (hunk 3 above)
Comment (miguelhenaoa) [reply src/scss/buttons.scss]: Done!

**Review by TaloMontes — DISMISSED** [review]

File: src/components/Input/styles.module.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+.form-alert {
+  color: $red;
+  margin-top: 2px;
+}
+
+.input-content {
+  margin-bottom: 16px;
+}
```
Comment (alejobh) [inline src/components/Input/styles.module.scss:26]: add EOF

File: src/constants/regex.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const regexEmail = /\S+@\S+\.\S+/;
```
Comment (alejobh) [inline src/constants/regex.ts:1]: you can name this file forms.ts

File: src/scss/variables/_colors.scss [hunk 6]
```diff
@@ -3,3 +3,8 @@ $black: #000;
 $red: #F00;
 $blue: #00F;
 $gray: #CCC;
+
+$cerulean: #00ADEE;
+$wild-sand:#F4F4F4;
+$earls-green: #BED23C;
+$silver-chalice: #A8A8A8;
```
Comment (alejobh) [inline src/scss/variables/_colors.scss:10]: Add EOF

File: src/scss/forms.scss [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    left: 0;
+    position: absolute;
+    width: 100%;
+  }
+}
+
+.form-alert {
+  color: $red;
+  margin-top: 2px;
+}
```
Comment (alejobh) [inline src/scss/forms.scss:22]: this is not used

File: src/scss/forms.scss [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  color: $black;
+  font-size: 16px;
+  height: 22px;
+  line-height: 24px;
+  margin-bottom: 8px;
+}
+
+.input-content {
+  margin-bottom: 16px;
+}
```
Comment (alejobh) [inline src/scss/forms.scss:42]: this is not used

File: src/scss/forms.scss [hunk 9]
```diff
@@ -0,0 +1,42 @@
+@import "./variables/colors";
```
Comment (alejobh) [inline src/scss/forms.scss:1]: i think all of this file does not need to be a global style

File: src/scss/forms.scss [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  &::after {
+    background-color: $silver-chalice;
+    bottom: 0;
+    content: "";
+    height: 2px;
+    left: 0;
+    position: absolute;
+    width: 100%;
+  }
+}
```
Comment (alejobh) [inline src/scss/forms.scss:17]: this is the only class used, and i think login and register screens looks the same.
What you can do is a presentational Wrapper component that receives a children. Then create a SignupForm and LoginForm components to be the children of this wrapper component.
For now and this PR this is ok. But refactor when facing login screen

**Review by alejobh — CHANGES_REQUESTED** [review]
Nice work! just a few comments

File: src/scss/forms.scss (hunk 7 above)
Comment (miguelhenaoa) [reply src/scss/forms.scss:22]: Removed!

File: src/scss/forms.scss (hunk 8 above)
Comment (miguelhenaoa) [reply src/scss/forms.scss:42]: Removed!

File: src/scss/forms.scss (hunk 10 above)
Comment (miguelhenaoa) [reply src/scss/forms.scss:17]: I will keep this in mind for when I make that screen

File: src/components/FormCard/index.tsx [hunk 11]
```diff
@@ -1,12 +1,16 @@
...
+import React, { ReactNode } from 'react';
 
 import styles from './styles.module.scss';
 
-function FormCard(props: any) {
+interface FormCardProps {
+  children: ReactNode;
+}
+
+function FormCard({ children }: FormCardProps) {
```
Comment (alejobh) [inline src/components/FormCard/index.tsx:9]: we usually name component's interface `Props`, Fix in the next PR

**Review by alejobh — APPROVED** [review]
Nice work!

File: src/components/FormCard/index.tsx (hunk 11 above)
Comment (miguelhenaoa) [reply src/components/FormCard/index.tsx:9]: Ok, I will keep in mind for the next cards.

**Pull request merged**